lib/util/index.js

```diff
@@ -483,30 +483,55 @@ function checkIfSdpSemanticsIsSupported() {
 function hidePrivateProperties(object) {
   Object.getOwnPropertyNames(object).forEach(name => {
     if (name.startsWith('_')) {
-      const descriptor = Object.getOwnPropertyDescriptor(object, name);
-      descriptor.enumerable = false;
-      Object.defineProperty(object, name, descriptor);
+      hideProperty(object, name);
     }
   });
 }
 
 /**
- * Creates a new subclass which, in the constructor, sets all
- * underscore-prefixed properties non-enumerable. This is useful for patching up
- * classes like EventEmitter which may set properties like `_events`.
+ * Creates a new subclass which, in the constructor, sets all underscore-prefixed
+ * properties and the given public properties non-enumerable. This is useful for
+ * patching up classes like EventEmitter which may set properties like `_events`
+ * and `domain`.
  * @param {Function} klass
+ * @param {Array<string>} props
  * @returns {Function} subclass
  */
-function hidePrivatePropertiesInClass(klass) {
+function hidePrivateAndCertainPublicPropertiesInClass(klass, props) {
   // NOTE(mroberts): We do this to avoid giving the class a name.
   return class extends klass {
     constructor(...args) {
       super(...args);
       hidePrivateProperties(this);
+      hidePublicProperties(this, props);
     }
   };
 }
 
+/**
+ * Hide a property of an object.
+ * @param {object} object
+ * @param {string} name
+ */
+function hideProperty(object, name) {
+  const descriptor = Object.getOwnPropertyDescriptor(object, name);
+  descriptor.enumerable = false;
+  Object.defineProperty(object, name, descriptor);
+}
+
+/**
+ * Hide the given public properties of an object.
+ * @param {object} object
+ * @param {Array<string>} props
+ */
+function hidePublicProperties(object, props) {
+  props.forEach(name => {
+    if (object.hasOwnProperty(name)) {
+      hideProperty(object, name);
+    }
+  });
```

Comment:
I thought the issue was that the Browserify EventEmitter sets a property on `prototype`. This no longer checks prototype, though.

Reply:
@markandrus I looked at the code, and EventEmitter's constructor explicitly sets an instance variable `domain` to `null`. So we have to hide it in the constructor.